-1 
Tested at: af4d2f90585ad5289cf169d11643e9c9e077324a
 I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 8 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>Selected class -> heppy::IsolationComputer for ROOT: heppy::IsolationComputer
Selected class -> heppy::PdfWeightProducerTool for ROOT: heppy::PdfWeightProducerTool
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-15-1100/src/Geometry/HGCalCommonData/src/HGCalGeometryMode.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-15-1100/src/Geometry/HGCalCommonData/src/HGCalParameters.cc 
In file included from /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-15-1100/src/Geometry/HGCalCommonData/src/HGCalDDDConstants.cc:1:
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-15-1100/src/Geometry/HGCalCommonData/interface/HGCalDDDConstants.h:87:27: error: constexpr variable 'tan30deg_' must be initialized by a constant expression
  constexpr static double tan30deg_ = std::sqrt(3.0)/3.0;
                          ^           ~~~~~~~~~~~~~~~~~~
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-15-1100/src/Geometry/HGCalCommonData/interface/HGCalDDDConstants.h:87:39: note: non-constexpr function 'sqrt' cannot be used in a constant expression
  constexpr static double tan30deg_ = std::sqrt(3.0)/3.0;
                                      ^
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12967/10546/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/a6f59e5be8f410715fdac0451fd4b57ab4430aa9
https://github.com/cms-sw/cmssw/commit/24b9332a4e3cf4f1d621dfc441cec0f3f13e36f6
https://github.com/cms-sw/cmssw/commit/795cbf441a611b5447f750efb73e220c28451e1f
https://github.com/cms-sw/cmssw/commit/cd4c10ef22f00ba0a8673126e6a68c3508a581b4
https://github.com/cms-sw/cmssw/commit/ede07a8f1d000471984fe12e39cafda4553322ec
https://github.com/cms-sw/cmssw/commit/f84cb95bab8bc6df01fe56916df4d6d0cba9992c
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12967/10546/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12967/10546/git-merge-result

--- a/Geometry/HGCalCommonData/interface/HGCalDDDConstants.h
+++ b/Geometry/HGCalCommonData/interface/HGCalDDDConstants.h
@@ -84,7 +84,7 @@ private:
   bool waferInLayer(int wafer, int lay) const;
 
   const HGCalParameters* hgpar_;
-  constexpr static double tan30deg_ = std::sqrt(3.0)/3.0;
+  constexpr static double tan30deg_ = 0.5773502693;
   double                 rmax_;
   HGCalGeometryMode      mode_;
 };

--- a/SLHCUpgradeSimulations/Configuration/python/combinedCustoms.py
+++ b/SLHCUpgradeSimulations/Configuration/python/combinedCustoms.py
@@ -54,6 +54,18 @@ def cust_2023HGCal_common(process):
         process.mix.digitizers.hgceeDigitizer=process.hgceeDigitizer
         process.mix.digitizers.hgchebackDigitizer=process.hgchebackDigitizer
         process.mix.digitizers.hgchefrontDigitizer=process.hgchefrontDigitizer
+        # update the HCAL Endcap for BH geom.
+        newFactors = cms.vdouble(
+            210.55, 197.93, 186.12, 189.64, 189.63,
+            189.96, 190.03, 190.11, 190.18, 190.25,
+            190.32, 190.40, 190.47, 190.54, 190.61,
+            190.69, 190.83, 190.94, 190.94, 190.94,
+            190.94, 190.94, 190.94, 190.94, 190.94,
+            190.94, 190.94, 190.94, 190.94, 190.94,
+            190.94, 190.94, 190.94, 190.94, 190.94,
+            190.94, 190.94, 190.94, 190.94, 190.94)
+        process.mix.digitizers.hcal.he.samplingFactors = newFactors
+        process.mix.digitizers.hcal.he.photoelectronsToAnalog = cms.vdouble([10.]*len(newFactors))
         # Also need to tell the MixingModule to make the correct collections available from
         # the pileup, even if not creating CrossingFrames.
         process.mix.mixObjects.mixCH.input.append( cms.InputTag("g4SimHits",process.hgceeDigitizer.hitCollection.value()) )

